eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/RequestPathPatternExtensions.cs

```diff
@@ -33,6 +35,11 @@ public static IReadOnlyList<ValueExpression> PopulateArguments(
                 {
                     arguments.Add(Convert(contextualParameter.BuildValueExpression(idProperty), typeof(string), parameter.Type));
                 }
+                //Find matching parameter from pathFieldsParameters if enclosing type is ResourceCollectionClientProvider
+                else if (enclosingType is ResourceCollectionClientProvider collectionProvider && collectionProvider.TryGetPrivateFieldParameter(parameter, out var matchingField) && matchingField != null)
```

Comment:
this looks weird.
But it is fine here because this method should not be here in the first place. We will refactor it in the future.